src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -990,12 +1009,21 @@ public StringBuilder Append(char value)
             }
             else
             {
-                Append(value, 1);
+                AppendWithExpansion(value);
             }
 
             return this;
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
I suspect we would not need to annotate this if we allowed a bigger inline budget....

Reply:
Not sure what you mean, we don't want this to be inlined because its a rare path and it would just unnecessarily bloat the caller of Append(char)